### ISSUE TITLE
There's a ton of people with the same issues, and nobody seems to want to help.

### ISSUE THREAD

**Issue by Murderface1982** [ramapcsx2/gbs-control#509]
I've sifted through countless issue threads, can those of us who solved this issue please just give a clear concise explanation of what exactly is wrong with the GUI when it is unresponsive?  I cannot access any resolutions on the web interface, I can't change much of anything besides the toggle switches.   Please, thank you.

**Comment by israpps** [comment]
I've been using my GBS Control without issues for a year.

U updated the firmware about 4 months ago and still 0 issues with the UI

**Comment by taylor-madeak** [comment]
To put it concisely: GBS-C is too busy _doing its job_ (upscaling a video signal) and doesn't have any available overhead for the web UI.

This one puzzled me at first, too. But once I noticed that the UI became responsive again after I turned off the incoming video signal, it immediately made sense to me. These devices aren't exactly supercomputers, they have limitations.

**Comment by Murderface1982** [comment]
> To put it concisely: GBS-C is too busy _doing its job_ (upscaling a video signal) and doesn't have any available overhead for the web UI.
> 
> This one puzzled me at first, too. But once I noticed that the UI became responsive again after I turned off the incoming video signal, it immediately made sense to me. These devices aren't exactly supercomputers, they have limitations.


I thought possibly the exact same thing, so i tried it with no signal, my PC was turned off and so was my TV, I'm still getting a unresponsive UI.

**Comment by YoImLee** [comment]
> To put it concisely: GBS-C is too busy _doing its job_ (upscaling a video signal) and doesn't have any available overhead for the web UI.
> 
> This one puzzled me at first, too. But once I noticed that the UI became responsive again after I turned off the incoming video signal, it immediately made sense to me. These devices aren't exactly supercomputers, they have limitations.

This is usually because of wifi interference between the ESP and the GBS. See here: https://ramapcsx2.github.io/gbs-control/Wiki/various-notes.html

As for none of the UI buttons doing anything, this happens sometimes on a fresh install. In my experience saving a preset can make the web UI responsive again.

**Comment by invaderlex** [comment]
because a ton of people fixed their wifi problem moving stuff. Where is your ESP mounted?

**Comment by 4lexei** [comment]
It's a bit unfair to say that we have no intention of helping. It's a free project and donations to Rama are voluntary, even he doesn't get much for his hard work.

The discussion about wifi connectivity issues have been discussed from the beginning. The solutions  have always been documented: change the heatsink, use other network channels or boost the antenna. What worked best for me was to design a box in 3D and place the ESP on the right side. Previously I used to install it vertically as Voultar showed in his video. But for me, in my particular case, it works better as I show in my photo


![BA16D2C7-9E2D-4E80-B7CD-1D2F609356F4](https://github.com/ramapcsx2/gbs-control/assets/34664444/8d8d7746-568e-4e1a-a5d0-b67938a6b743)

**Comment by ramapcsx2** [comment]
Yep, it can be very small changes like that. WiFi wavelength is small, and all it takes is to remove the ESP antenna from a particular set of interference.

**Comment by invaderlex** [comment]
> It's a bit unfair to say that we have no intention of helping. It's a free project and donations to Rama are voluntary, even he doesn't get much for his hard work.
> 
> The discussion about wifi connectivity issues have been discussed from the beginning. The solutions have always been documented: change the heatsink, use other network channels or boost the antenna. What worked best for me was to design a box in 3D and place the ESP on the right side. Previously I used to install it vertically as Voultar showed in his video. But for me, in my particular case, it works better as I show in my photo
> 
> ![BA16D2C7-9E2D-4E80-B7CD-1D2F609356F4](https://user-images.githubusercontent.com/34664444/282820303-8d8d7746-568e-4e1a-a5d0-b67938a6b743.jpeg)

YES! mounting the ESP over the GBS is like mounting an antenna over a big sheet of cooper. The solder mask becomes a shield. Mine is also on the same spot as your and I have no connection problems.

**Comment by ktElwood** [comment]
Try to access your ESP32 while there is no video input on the GBS82X0 or the GBS8200 itself isn't powered

I use an old phone to connect to the GBS-Control Wifi  in AP mode.

Once it's connected I have no issues using the GUI on the websever while my Wii/Ps1/Ps2  etc. are running through the GBS8200.

**Comment by Lewinsky123** [comment]
Has anyone compiled the sketch to work off an ESP32-S2-WROVER, or any chip that has more power to it?

**Comment by geekboo** [comment]
> Has anyone compiled the sketch to work off an ESP32-S2-WROVER, or any chip that has more power to it?

I would also like to know the ESP32 has a much more powerful processor and Im struggling with my ESP8266, I cant change resolution through the GUI, its connected to Wi-Fi but the buttons simply do nothing.

**Comment by Kwakx** [comment]
If anyone has a problem with Wi-Fi, try my fork. Since I made the changes, I have had zero problems with Wi-Fi and websocket.

https://github.com/Kwakx/gbs-control-nx/releases

Everything is built and upload via the platformio. I also plan to prepare compatibility with ESP32, but first I need to buy the original GBS 8220.

**Comment by ramapcsx2** [comment]
@Kwakx: I read that your core improvement is a "forced 802.11g" mode, along with monitoring and auto reconnect stuff, is that right?
The core issue is that the ASIC generates EMI (noise) that randomly falls within with Wi-Fi frequency bands, blocking all communication. It could be that the forced g mode helps with that, but it remains to be seen how it works for others. Good on you for working on this though! :)

**Comment by Kwakx** [comment]
@ramapcsx2 I started by updating all libraries and the platform to the latest version, and even then I noticed an improvement in the performance of the UI/websocket, and in my case, the Wi-Fi worked until the next or random restart.

Adding a few delays, status clears, etc., along with forced 802.11g mode, brought the final result. Auto reconnect is more of an extra feature.

Additionally, the update to Arduino core v3 required a change in the approach to saving credentials and requires the use of WiFi.persistent( ). 

https://arduino-esp8266.readthedocs.io/en/latest/esp8266wifi/generic-class.html
https://github.com/esp8266/Arduino/pull/7902

**Comment by ramapcsx2** [comment]
Ok, sounds good. Some core driver changes in the past years might very well have fixed it :)

**Comment by Kwakx** [comment]
https://github.com/Kwakx/gbs-control-nx/releases/tag/v1.2.2

I released another patch addressing these issues. Websocket is now so stable that you can spam commands in WebUI and everything is processed without interrupting the WS connection.
